src/main/java/org/ballerinax/datamapper/DataMapperStructureVisitor.java

```diff
@@ -0,0 +1,139 @@
+    private Map<String, String> structureMap;
+
+    public DataMapperStructureVisitor(Map structure){
+        structureMap = structure;
+    }
+
+    @Override
+    public void visit(BLangTypeDefinition typeDefinition) {
+        for (Iterator<BLangSimpleVariable> iterator = ((BLangRecordTypeNode) typeDefinition.typeNode).
+            fields.iterator(); iterator.hasNext(); ) {
+            BLangSimpleVariable field = iterator.next();
+
+            if (field.getKind() == NodeKind.VARIABLE) {
+                field.accept(this);
+            }
+        }
+    }
+
+    @Override
+    public void visit(BLangTypeInit varRefExpr) {
+        List<BLangExpression> exprList = varRefExpr.argsExpr;
+        ArrayList<BVarSymbol> list = (ArrayList<BVarSymbol>) ((BObjectTypeSymbol) varRefExpr.type.tsymbol).
+                                    initializerFunc.symbol.params;
+        int count = 0;
+        for (Iterator<BLangExpression> iterator = exprList.iterator(); iterator.hasNext(); ) {
+            BLangExpression expr = iterator.next();
+            if (expr.getKind() == NodeKind.LITERAL) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.NAMED_ARGS_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR) {
+                expr.accept(this);
+            }
+            count += 1;
+        }
+    }
+
+    @Override
+    public void visit(BLangNamedArgsExpression varRefExpr) {
+        BLangExpression expr = (BLangExpression) varRefExpr.getExpression();
+        if (expr.getKind() == NodeKind.LITERAL) {
+            expr.accept(this);
+        } else if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+            expr.accept(this);
+        } else if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR) {
+            expr.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangListConstructorExpr listConstructorExpr) {
+        List<BLangExpression> exprList = listConstructorExpr.getExpressions();
+
+        for (BLangExpression expr : exprList) {
+            if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                //Note: Here we do not need to travers further because we know that this is an array
+                //and the type is known in advance in this method using listConstructorExpr.type
+            }
+        }
+    }
+
+    @Override
+    public void visit(BLangSimpleVariable varNode) {
+        structureMap.put(varNode.symbol.name.toString(), varNode.symbol.type.toString());
+    }
+
+    @Override
+    public void visit(BLangRecordLiteral.BLangRecordKeyValueField recordKeyValue) {
+        if (recordKeyValue.valueExpr.getKind() == NodeKind.LITERAL) {
```

Comment:
IMO a switch case would be more readable `switch(recordKeyValue.valueExpr.getKind()){...}`

Reply:
Yes, the readability can be improved using the switch statement. I updated the code to follow this approach.